### ISSUE TITLE
Resources does not match result

### ISSUE THREAD

**Issue by nesl247** [pulumi/pulumi-gcp#171]
I just attempted to update my GKE cluster. The name of the nodepools changed, as well as the version.

The upgrade actually failed, but the `Resources` section at the end does not reflect it properly.

```
     Type                       Name                Status                   Info
     pulumi:pulumi:Stack        kubernetes-staging  **failed**               1 error
 ~   ├─ gcp:container:Cluster   gke                 updated                  [diff: ~minMasterVersion]
 +-  ├─ gcp:container:NodePool  web                 **replacing failed**     [diff: ~name,version]; 1 error
 +-  └─ gcp:container:NodePool  cli                 **replacing failed**     [diff: ~name,version]; 1 error

Diagnostics:
  gcp:container:NodePool (cli):
    error: Plan apply failed: deleting urn:pulumi:staging::kubernetes::gcp:container/nodePool:NodePool::cli: Error deleting NodePool: googleapi: Error 400: Operation operation-1561562702727-c8a7b057 is currently upgrading cluster linio. Please wait and try again once it is done., failedPrecondition

  gcp:container:NodePool (web):
    error: Plan apply failed: deleting urn:pulumi:staging::kubernetes::gcp:container/nodePool:NodePool::web: Error deleting NodePool: googleapi: Error 400: Operation operation-1561562702727-c8a7b057 is currently upgrading cluster linio. Please wait and try again once it is done., failedPrecondition

  pulumi:pulumi:Stack (kubernetes-staging):
    error: update failed
```

```
Resources:
    ~ 1 updated
    +-2 replaced
    2 changes. 6 unchanged
```

The resources should not say the 2 resources were replaced, because it did not succeed.

On top of this, why was it attempting to replace anything when GKE itself still shows the masters as upgrading? The NodePools did actually get created though, it failed to remove the old ones due to the master upgrade.

Something is definitely wrong here.

**Comment by lukehoban** [comment]
I believe we report at the end the operations that were "started".  I'm not sure it would be striclty better to report just what ended - as operations that started but failed may have had an impact that matters on the cloud provider.  We should probably be clear on which operations completed successfully and which failed in this summary.

> On top of this, why was it attempting to replace anything when GKE itself still shows the masters as upgrading? The NodePools did actually get created though, it failed to remove the old ones due to the master upgrade.

That sounds like a separate issue - and if needed could open an issue on this.  You may want to look at https://github.com/pulumi/pulumi-gcp/issues/88 which catalogs a few issues related to unintentended replacements for GKE resources (as well as some debugging techniques you could use to get more details on what's really causing these replacements).

**Comment by nesl247** [comment]
I agree that it should be explicit about what started and what completed. That makes a lot of sense.

Regarding the other thing, I'll open up an issue if it happens again. The replacement was actually expected.
